server/server.go

```diff
@@ -133,6 +135,35 @@ func (s *Server) isUnixSocket() bool {
 	return s.cfg.Socket != ""
 }
 
+func (s *Server) forwardUnixSocketToTCP(socket string, addr string) {
+	if sock, err := net.Listen("unix", socket); err == nil {
+		log.Infof("Server redirecting [%s] to [%s]", socket, addr)
+		for {
+			if uconn, err := sock.Accept(); err == nil {
+				log.Infof("server socket forwarding from [%s] to [%s]", socket, addr)
+				go s.handleForwardedConnection(uconn, addr)
+			} else {
+				log.Warningf("server failed to forward from [%s] to [%s], err: %s", socket, addr, err)
+			}
+		}
+	} else {
+		log.Fatalf("err: %s", err) // in use?
+	}
+}
+
+func (s *Server) handleForwardedConnection(uconn net.Conn, addr string) {
```

Comment:
Rather than logging warnings, I would return an error and have the caller deal with it (the caller can log or retry).